include/sys/util.h

```diff
@@ -110,6 +110,10 @@ size_t ARRAY_SIZE(T(&)[N]) { return N; }
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 #endif
 
+#ifdef __cplusplus
```

Comment:
Please move this block up to below the `#include` statements.  This will help avoid problems if somebody adds a new declaration earlier in the file than `is_power_of_two`.

Reply:
Ok, fixed this.  But I moved it to the existing cplusplus conditional (there is a cpp template in the header).

Reply:
I think it would be better to put `extern "C"` where I proposed originally, then put an `extern "C++"` around the template definition within the `defined(__cplusplus)` region.  It's confusing to see these things anywhere other than the start and end of the region they apply to.

Does that work?

Reply:
Ok.  Fixed.